apps/services/schemas.py

```diff
@@ -13,9 +13,73 @@
 )
 
 
-class CreateUserRequestIn(BaseInSchema):
-    service_id: int
+class RequestForHostelAccommodationIn(BaseInSchema):
+    rector_first_name: str
+    rector_middle_name: str = None
+    rector_last_name: str
+    student_first_name: str
+    student_middle_name: str = None
+    student_last_name: str
+    speciality_code: int
+    speciality_name: str
+    course: int
+    faculty_name: str
+    educ_level: str
+    comment: str = None
+
+    @root_validator
+    def validate_rector_and_student_names(cls, values: str) -> str:
+        names = [
+            "rector_first_name",
+            "rector_last_name",
+            "student_first_name",
+            "student_last_name",
+        ]
+        for name in names:
+            if not values.get(name).istitle():
+                raise ValueError(
+                    f"{name.replace('_', ' ').capitalize()} first letter must be uppercase"
+                )
+        return values
+
+    @validator("student_middle_name")
+    def validate_student_middle_name(cls, value: str) -> Optional[str]:
+        if value:
+            if not value.istitle():
+                raise ValueError("Middle name must be uppercase")
+            return value
+        return value
+
+    @validator("rector_middle_name")
+    def validate_rector_middle_name(cls, value: str) -> Optional[str]:
+        if value:
+            if not value.istitle():
+                raise ValueError("Middle name must be uppercase")
+            return value
+        return value
+
+    @validator("educ_level")
+    def validate_education_level(cls, value: str) -> str:
+        if value.upper() not in ["B", "M"]:
+            raise ValueError("Wrong education level value. Should be 'B' or 'M'")
+        return value
+
+    @validator("course")
+    def validate_course(cls, value: int) -> int:
+        if value not in [1, 2, 3, 4, 5, 6]:
+            raise ValueError(f"Wrong '{value}' number. This course doesn't exist")
+        return value
+
+
+class RequestForHostelAccommodationOut(BaseOutSchema):
```

Comment:
It's better to write titles, descriptions and provide a Field instance for every field in schemas.

Reply:
ok